modules/ROOT/pages/_partials/partner-and-customer-guide.adoc

```diff
@@ -45,12 +45,11 @@ To decouple the minimum Mule version when running tests, add the `-Dmunit.jvm` p
 Before upgrading and testing your custom connector for Java 17, you must be familiar with the following considerations:
 
 * Target compilation level
+From Mule 4.9.0 and onwards, connectors can be compiled with any Java versions providing it can generate bytecode the proper version.
```

Comment:
is this grammar ok? I would rephase as `From Mule 4.9.0 and onwards, any Java SDK project can be compiled for Java 17 language level or lower. Any additional dependencies must also be compiled for a language level equal or lower than 17`

Notice that:

* This now includes modules, not only connectors
* It's not making promisises about supporting Java 21 compilation at day zero, since Hyperforce migration might put that at risk